src/app/zap-templates/templates/chip/helper.js

```diff
@@ -445,6 +446,130 @@ async function chip_endpoint_clusters(options)
   return templateUtil.collectBlocks(clusters, options, this);
 }
 
+/**
+ * Helper checks if the type for the bitmap is BitFlags. This generally includes
+ * all bitmaps apart from
+ * bitmap8/16/32(generally defined in types.xml)
+ * example:
+ * {{#if_is_strongly_typed_bitmap type}}
+ * strongly typed bitmap
+ * {{else}}
+ * not a strongly typed bitmap
+ * {{/if_is_strongly_typed_bitmap}}
+ *
+ * @param {*} type
+ * @returns Promise of content.
+ */
+async function if_is_strongly_typed_bitmap(type, options)
+{
+  let packageId = await templateUtil.ensureZclPackageId(this)
+  let bitmap
+  if (type && typeof type === 'string')
+  {
+    bitmap = await queryZcl.selectBitmapByName(this.global.db, packageId, type)
+  }
+  else
+  {
+    bitmap = await queryZcl.selectBitmapById(this.global.db, type)
+  }
+
+  if (bitmap) {
+    let a = await queryZcl.selectAtomicType(this.global.db, packageId, bitmap.name)
+    if (a)
+    {
+      return options.inverse(this)
+    }
+    else
+    {
+      return options.fn(this)
+    }
+  }
+  return options.inverse(this)
+}
+
+/**
+ * Helper to a handlebar helper function which checks if an enum is a strongly
+ * typed enum or not. This generally includes all enums apart from
+ * enum8/16/32(generally defined in types.xml)
+ *
+ * * example for if_is_strongly_typed_enum:
+ * {{#if_is_strongly_typed_enum type}}
+ * strongly typed enum
+ * {{else}}
+ * not a strongly typed enum
+ * {{/if_is_strongly_typed_enum}}
+ *
+ * @param {*} type
+ * @param {*} options
+ * @param {*} context
+ * @returns Promise of content.
+ */
+async function if_is_strongly_typed_enum_common(type, options, context)
+{
+  let packageId = await templateUtil.ensureZclPackageId(context)
+  let enumRes
+  // Retrieving the enum from the enum table
+  if (type && typeof type === 'string')
+  {
+    enumRes = await queryZcl.selectEnumByName(context.global.db, type, packageId)
+  }
+  else
+  {
+    enumRes = await queryZcl.selectEnumById(context.global.db, type)
+  }
+
+  // Checking if an enum is atomic. If an eunm is not atomic then the enum
+  // is a strongly typed enum
+  if (enumRes) {
+    let a = await queryZcl.selectAtomicType(context.global.db, packageId, enumRes.name)
+    if (a)
+    {
+      return options.inverse(context)
```

Comment:
Again, for the third time: should document that if the enum has an atomic type that means it's a weakly-typed enum.